src/OmniSharp.Roslyn.CSharp/Services/Structure/CodeStructureService.cs

```diff
@@ -0,0 +1,411 @@
+                    return CodeElementKinds.Namespace;
+                case SyntaxKind.OperatorDeclaration:
+                case SyntaxKind.ConversionOperatorDeclaration:
+                    return CodeElementKinds.Operator;
+                case SyntaxKind.PropertyDeclaration:
+                    return CodeElementKinds.Property;
+                case SyntaxKind.StructDeclaration:
+                    return CodeElementKinds.Struct;
+                default:
+                    return CodeElementKinds.Unknown;
+            }
+        }
+
+        private static string GetAccessibility(ISymbol symbol)
+        {
+            switch (symbol.DeclaredAccessibility)
+            {
+                case Accessibility.Public:
+                    return CodeElementAccessibilities.Public;
+                case Accessibility.Internal:
+                    return CodeElementAccessibilities.Internal;
+                case Accessibility.Private:
+                    return CodeElementAccessibilities.Private;
+                case Accessibility.Protected:
+                    return CodeElementAccessibilities.Protected;
+                case Accessibility.ProtectedOrInternal:
+                    return CodeElementAccessibilities.ProtectedInternal;
+                case Accessibility.ProtectedAndInternal:
+                    return CodeElementAccessibilities.PrivateProtected;
+                default:
+                    return null;
+            }
+        }
+
+        private static Range CreateRange(TextSpan span, SourceText text)
+        {
+            var startLine = text.Lines.GetLineFromPosition(span.Start);
+            var endLine = text.Lines.GetLineFromPosition(span.End);
+
+            return new Range
+            {
+                Start = new Point { Line = startLine.LineNumber, Column = span.Start - startLine.Start },
+                End = new Point { Line = endLine.LineNumber, Column = span.End - endLine.Start }
+            };
+        }
+
+        private static void AddRanges(CodeElement.Builder builder, TextSpan attributesSpan, TextSpan fullSpan, SourceText text)
+        {
+            if (attributesSpan != default)
+            {
+                builder.AddRange(CodeElementRangeNames.Attributes, CreateRange(attributesSpan, text));
+            }
+
+            if (fullSpan != default)
+            {
+                builder.AddRange(CodeElementRangeNames.Full, CreateRange(fullSpan, text));
+            }
+        }
+
+        private void AddSymbolProperties(ISymbol symbol, CodeElement.Builder builder)
+        {
+            var accessibility = GetAccessibility(symbol);
+            if (accessibility != null)
+            {
+                builder.AddProperty(CodeElementPropertyNames.Accessibility, accessibility);
+            }
+
+            builder.AddProperty(CodeElementPropertyNames.Static, symbol.IsStatic);
+
+            foreach (var propertyProvider in _propertyProviders)
```

Comment:
I like the extensibility, much cleaner than the existing strange bolt-on 🙂

Reply:
I was not a fan of the bolt on. :smile: